src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -772,7 +783,7 @@ private static void EmitSimplifiedGo(IndentedTextWriter writer, RegexMethod rm,
 
             // Emit failure
             writer.WriteLine("// No match");
-            MarkLabel(originalDoneLabel, !expressionHasCaptures);
+            MarkLabel(originalDoneLabel);
```

Comment:
Small thing, but I believe that with this change and because the default is to emit semicolon, won't this result in code like:
```c#
// No match
NoMatch:;
// ...
```

I believe that in this case you always want to pass `false` to MarkLabel no?

Reply:
> won't this result in code like

Yes

> I believe that in this case you always want to pass false to MarkLabel no?

No :) If `expressionHasCaptures` is false, then the label will be the last thing in the method, and not including a semicolon will fail to compile.  I could put back the check for expressionHasCaptures in order to make it conditional... just started to seem like more work than it's worth.

Reply:
oh got it, makes sense. I think it is fine, it might look a tiny bit weird for folks to see empty statements when using captures, but that should be fine.